src/subdomains/core/buy-crypto/process/services/buy-crypto.service.ts

```diff
@@ -349,6 +363,44 @@ export class BuyCryptoService {
     });
   }
 
+  async refundCryptoInput(
+    buyCrypto: BuyCrypto,
+    chargebackDto: TransactionRefundDto,
+    chargebackAmount?: number,
+  ): Promise<void> {
+    if (!chargebackDto.refundAddress && !chargebackDto.refundUserId)
+      throw new BadRequestException('You have to define a chargebackAddress');
+
+    const refundUser = chargebackDto.refundUserId
+      ? await this.userService.getUser(chargebackDto.refundUserId, { userData: true })
+      : await this.userService.getUserByAddress(chargebackDto.refundAddress, { userData: true });
+
+    if (buyCrypto.userData.id !== refundUser.userData.id)
+      throw new ForbiddenException('You can only refund to your own addresses');
+    if (!refundUser.blockchains.includes(buyCrypto.cryptoInput.asset.blockchain))
+      throw new BadRequestException('You can only refund to a address on the origin blockchain');
+    if (buyCrypto.chargebackAllowedDate || buyCrypto.chargebackDate || buyCrypto.chargebackIban)
+      throw new BadRequestException('Transaction is already returned');
+    if (buyCrypto.amlCheck !== CheckStatus.FAIL || buyCrypto.outputAmount)
+      throw new BadRequestException('Only failed transactions are refundable');
+    if (chargebackAmount && chargebackAmount > buyCrypto.inputAmount)
+      throw new BadRequestException('You can not refund more than the input amount');
```

Comment:
Can we somehow share these checks with BuyFiat?

Reply:
Hm yeah if we find a good spot for it we could do that 👍 
Do you have a fitting spot/service or should make an CryptoInput Share Service or something like that? 🤔

Reply:
Maybe a helper?